### ISSUE TITLE
there no method for docker stats?

### ISSUE THREAD

**Issue by crazywen** [spotify/docker-client#172]
there no method for docker stats?
why?
Get container stats based on resource usage
GET /containers/(id)/stats

This endpoint returns a live stream of a container's resource usage statistics.

**Comment by davidxia** [comment]
@crazywen I guess we just haven't needed it yet. We're definitely open to PRs though. Let me know if you need some pointers on how to get started.

**Comment by crazywen** [comment]
I think ,just retrun the stats response to a pojo  is ok! but I don't  know how to handle the 'live stream'.

**Comment by davidxia** [comment]
@crazywen I looked into this briefly. The HTTP response that comes back is chunked transfer encoding (http://en.wikipedia.org/wiki/Chunked_transfer_encoding). I haven't had time to figure out how to consume that yet. I tried to mimic `LogStream` but didn't get that far.

**Comment by crazywen** [comment]
@davidxia  yeah,the response is a chunked;ant I resolved  by apache httpClient, with the response httpEntity,
```
@SuppressWarnings("finally")
	public ContainerStat getContainerStat(String uri, String containerId)
			throws IOException {
		CloseableHttpClient httpclient = HttpClients.createDefault();
		String stat = null;
		try {
			HttpGet httpget = new HttpGet(uri + "containers/" + containerId
					+ "/" + STAT_RESOURCES + "?stream=0");
			if (logger.isDebugEnabled()) {
				logger.debug(httpget.getRequestLine().toString());
			}
			RequestConfig requestConfig = RequestConfig.custom()
					.setSocketTimeout(1000).setConnectTimeout(1000).build();
			httpget.setConfig(requestConfig);
			CloseableHttpResponse response = httpclient.execute(httpget);
			HttpEntity entity = response.getEntity();
			InputStream is = entity.getContent();
			BufferedReader reader = new BufferedReader(
					new InputStreamReader(is));

			while (null != (stat = reader.readLine())) {
				// just get the first line response
				httpget.abort();
				break;
			}
		} finally {
			httpclient.close();
			if (StringUtil.isBlank(stat)) {
				return null;
			}
			return JsonUtil.fromJson(stat, ContainerStat.class);
		}
	}
```

**Comment by davidxia** [comment]
@crazywen I tried to get something going here using our existing LogStream class but it gets stuck on readFully() https://github.com/spotify/docker-client/tree/stats

**Comment by unguiculus** [comment]
I also started working on this and just found out I'm not the only one. I don't have anything functional yet, though.

**Comment by unguiculus** [comment]
@davidxia I'm not interested in streaming. I'd rather like to get one-time stats. I added a method to LogReader to try this out. It works but doesn't quite fit in there conceptually.

```java
public String nextLine() throws IOException {
    try (BufferedReader br = new BufferedReader(new InputStreamReader(stream))) {
      String line = br.readLine();
      log.debug("Stats line: {}", line);
      return line;
    }
  }
```

BTW, in Docker 1.7, there will be an additional parameter, so stats can be pulled once without streaming. Maybe we should just wait and then use the non-streaming version.

See https://github.com/docker/docker/pull/10766

URLs will look like this: `/containers/<container>/stats?stream=0`

**Comment by unguiculus** [comment]
I just created a pull request (https://github.com/spotify/docker-client/pull/197) for one-time stats support. I had to get this issue fixed to get it working: https://github.com/docker/docker/issues/13910. So, Docker 1.7.2-rc3 or higher is required.

Thanks.

**Comment by rohansingh** [comment]
Fixed by #197.
